src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -161,13 +183,48 @@ static void checkArg(string arg, string name)
                 for (int i = 0; i < iterations; i++)
                 {
                     logger.LogIterationHeader($"Iteration {i}");
-                    var iterationResult = RunIteration(setupProcHelper, TestProcess, cleanupProcHelper, logger);
+                    (bool Success, int Pid) iterationResult;
+                    if(metricType == MetricType.InnerLoop)
+                    {
+                        iterationResult = RunIteration(setupProcHelper, TestProcess, null, logger);
+                    }
+                    else
+                    {
+                        iterationResult = RunIteration(setupProcHelper, TestProcess, cleanupProcHelper, logger);
+                    }
                     if (!iterationResult.Success)
                     {
                         failed = true;
                         break;
                     }
                     pids.Add(iterationResult.Pid);
+
+                    if(metricType == MetricType.InnerLoop)
+                    {
+                        //Do some stuff to change the project
+                        logger.LogStepHeader("Inner Loop Setup");
+                        var innerLoopReturn = innerLoopProcHelper.Run();
+                        if(innerLoopReturn.Result != ProcessHelper.Result.Success)
+                        {
+                            failed = true;
+                            break;
+                        }
+                        else
+                        {
+                            var test = InnerLoopMarkerEventSource.GetSources();
+                            InnerLoopMarkerEventSource.Log.Split();
+
+                            logger.LogIterationHeader($"Iteration {i} - Diff");
+                            iterationResult = RunIteration(null, TestProcess, cleanupProcHelper, logger);
+                            if (!iterationResult.Success)
+                            {
+                                failed = true;
+                                break;
+                            }
+                            pids.Add(iterationResult.Pid);
+                            InnerLoopMarkerEventSource.Log.EndIteration();
```

Comment:
could this be just after `RunIteration`?

Reply:
oh, I guess it doesn't matter since these are just fences and not measurement spots.

Reply:
Yeah, I just use them as fences to let me know which side of the test we are in.